Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs

```diff
@@ -162,6 +189,161 @@ public override void Initialize()
+        {
+            // Nukies must wait 15 minutes after declaration of war to get on the station
+            var gameruleTime = _gameTiming.CurTime.Subtract(_declarationTime);
+            if (gameruleTime > _nukeopsRuleConfig.WarNukieArriveDelay)
+            {
+                return WarConditionStatus.WAR_READY;
+            }
+            return WarConditionStatus.WAR_DELAY;
+        }
+        else
+        {
+            if (_operativePlayers.Count < _nukeopsRuleConfig.WarMinCrewSize) return WarConditionStatus.NO_WAR_SMALL_CREW;
+            if (_leftOutpost) return WarConditionStatus.NO_WAR_SHUTTLE_DEPARTED;
+    
+            var gameruleTime = _gameTiming.CurTime.Subtract(_gameruleStartTime);
+            if (gameruleTime > _nukeopsRuleConfig.WarDeclarationTimeWindow) return WarConditionStatus.NO_WAR_TIMEOUT;
+    
+            return WarConditionStatus.YES_WAR;
+        }  
+    }
+    
+    private void OnFTLTravelAttempt(ShuttleConsoleFTLTravelAttemptEvent ev)
+    {
+        if (!RuleAdded)
+            return;
+        
+        if (_nukieOutpost == null)
+        {
+            return;
+        }
+        var map_outpost = _entityManager.GetComponent<TransformComponent>(_nukieOutpost.Value).MapID;
+        var map_shuttle = _entityManager.GetComponent<TransformComponent>(ev.Uid).MapID;
+
+        if (_declarationMade && map_outpost == map_shuttle)
+        {
+            var timeAfterDeclaration = _gameTiming.CurTime.Subtract(_declarationTime);
+            var timeRemain = _nukeopsRuleConfig.WarNukieArriveDelay.Subtract(timeAfterDeclaration);
+            if (timeRemain > TimeSpan.Zero)
+            {
+                ev.Cancel();
+                ev.Reason = Loc.GetString("war-ops-infiltrator-unavailable", ("minutes", timeRemain.Minutes), ("seconds", timeRemain.Seconds));
+            }
+        }
+    }
+
+    private void OnShuttleCallAttempt(CommunicationConsoleCallShuttleAttemptEvent ev)
+    {
+        if (!RuleAdded || _nukeopsRuleConfig.PreventShuttleInDelay == false)
+            return;
+        
+        if (_declarationMade)
+        {
+            var gameruleTime = _gameTiming.CurTime.Subtract(_declarationTime);
+            if (gameruleTime <= _nukeopsRuleConfig.WarNukieArriveDelay)
+            {
+                ev.Cancel();
+                ev.Reason = Loc.GetString("war-ops-shuttle-call-unavailable");
+            }
+        }
+    }
+
+    private void OnFTLTravel(ShuttleFTLTravelEvent ev)
+    {
+        var shuttle = ev.Shuttle.Owner;
+    	if (!RuleAdded)
+            return;
+
+        if (_nukieOutpost == null)
+        {
+            return;
+        }
+        var map_outpost = _entityManager.GetComponent<TransformComponent>(_nukieOutpost.Value).MapID;
+        var map_shuttle = _entityManager.GetComponent<TransformComponent>(shuttle).MapID;
+
+        if (map_outpost == map_shuttle)
+        {
+            _leftOutpost = true;
+            _warDeclaratorSystem.RefreshAllDeclaratorsUI();
```

Comment:
Just use a directed event because then you know which shuttle it is? Also the local variable formatting doesn't conform to C# style guide.

Reply:
I want to count every FTL travel at outpost map, not only the syndie shuttle. So if ops would build their own shuttle and use it for travel, then it still will be counted as leave from outpost